docs/book/src/clusterctl/commands/alpha-topology-plan.md

```diff
@@ -1,3 +1,172 @@
 # clusterctl alpha topology plan
 
-**Under Construction**
+The `clusterctl alpha topology plan` command can be used to dry run the topology reconciler using an input file.
+
+The input file could contain a new/modified Cluster, a new/modified ClusterClass and/or new/modified templates.
+
+The dry run output would list the Clusters and ClusterClasses that would be affected by the input. The output also lists the
+objects that will be created, updated and deleted of a target cluster.
+
+This command can be used with and without a management cluster. In case the command is used without a management cluster the input should have
+all the objects needed to perform the dry run. In case the command is run with a management cluster the dry run operation will look for objects 
+in the real cluster as needed.
+
+```shell
+clusterctl alpha topology plan -f input.yaml -o output/
+```
+
+> Note: This command can be used with or without a management cluster. In case the command is used without a management cluster the input should have
+> all the objects needed to perform the dry run. In case the command is run with a management cluster the dry run operation will look for objects 
+> in the management cluster as needed.
+
+## Dry run
+
+The operation is composed of the following steps:
+* Set the namespace on objects in the input with missing namespace.
+* Run the Defaulting and Validation webhooks on the Cluster and ClusterClass objects in the input.
+* Dry run the topology reconciler on the target cluster.
+* Capture all changes observed during reconciliation.
+
+<aside class="note">
+
+<h1>Set missing namespace</h1>
+
+The namespace used for objects with missing namespace follows this precedence order:
+* Value passed using `--namespace`
+* Current namespace defined in kubeconfig
+* Value `default`
+
+</aside>
+
+<aside class="note warning">
+
+<h1>Defaulting and Validation of objects</h1>
+
+Defaulting and validation is only performed on the Cluster and ClusterClass objects in the input. Objects of all other kinds in the input should 
+be fully valid and any required default values should be set. Missing values will not be defaulted.
```

Comment:
Should we add the tip that users can also apply those YAMLs if they have a mgmt cluster and then use the resulting objects as input for the command? (either by getting them or just using them directly from the cluster)

Reply:
I see your point. However, suggesting users to apply the YAMLs doesn't fit will here as the whole idea of the command is that users can try stuff without actually applying any changes to the real cluster.

Reply:
Yup makes sense